climakitae/data_export.py

```diff
@@ -0,0 +1,225 @@
+              " GB. This might take a while!")              
+              
+            
+    if ("NetCDF" in req_format):
+        print("Alright, exporting specified data to NetCDF. This might take a while - "+
+             "hang tight!")
+        data_to_export.to_netcdf(path_and_name+".nc")
+        
+    else:        
+        data_to_export = data_to_export.squeeze()
+        print("NOTE: Metadata will be saved to a separate file called "+
+              file_name+"_metadata.txt. Be sure to download it"+
+              " when you download your data!")       
+        metadata_to_file(data_to_export,file_name,req_format)# make metadata text file
+    
+        if ("CSV" in req_format):
+            print("WARNING: Exporting to CSV can take a long time,"+
+                         " and is not recommended for data with more than 2 coordinates."+
+                          " Please note that the data will be compressed via gzip. Even so,"+
+                 " inherent inefficiencies may result in a file which is too large to save here.")
+            print("Converting data...")
+            to_save = data_to_export.to_dataframe()        
+            csv_nrows = len(to_save.index)        
+            if (csv_nrows > excel_row_limit):
+                print("WARNING: Dataset exceeds Excel limit of "+
+                              str(excel_row_limit)+" rows.")        
+            print("Compressing data... This can take a bit...")
+            to_save.to_csv(path_and_name+".gz",compression='gzip')
+
+        elif ("GeoTIFF" in req_format):
+            if ('time' in data_to_export.dims):
+                print("NOTE: Saving time series as multiband raster in which "+
+                     "each band corresponds to a time step.")
+                print("See metadata file for more information.")
+            print("Saving as GeoTIFF...")
+            data_to_export.rio.to_raster(path_and_name+".tif")
+                   
+    return(print("Saved! You can find your file(s) in the panel to the left "+
+                "and download to your local machine from there." ))
+
+
+def metadata_to_file(ds,output_name,req_format):
+    """
+    Writes NetCDF metadata to a txt file so users can still access it 
+    after exporting to a CSV or GeoTIFF.
+    Does not appear to work for GeoTIFF.
+    """
+    if os.path.exists(output_name+"_metadata.txt"):
+        os.remove(output_name+"_metadata.txt")
+        
+    with open(output_name+"_metadata.txt", 'w') as f:
+        f.write('======== Metadata for '+req_format+' file '+output_name+' ========')
+        f.write('\n')
+        f.write('\n')
+        f.write('\n')
+        f.write('===== Global file attributes =====')
+        f.write('\n')
+        f.write('\n')
+        for att_keys,att_values in list(zip(ds.attrs.keys(),ds.attrs.values())):    
+            f.write(str(att_keys)+" : "+str(att_values))
+            f.write('\n')                       
```

Comment:
This part is nice. I see it in the metadata.txt that accompanies a .csv export, but I don't see something equivalent in the netCDF metadata if no transforms are performed (e.g. if you just retrieve and then export something).

Reply:
> Two big-picture comments --
> 
> 1. @bkg  has told me that it's not ideal to use print() statements in library code. I am not, however, sure what you should use instead to display your helpful little messages below the cell, or how big a deal this is.

Would love to hear the alternative. Google is giving me nothing. 

For reference, I tried the warnings package, but I disliked that it didn't just show my warning - it showed where in the function call I wrote that warning. Also, I'm not always outputting a warning - sometimes it's a success statement. 

> 2. If I supply the filename 'my_testfile.nc' then it writes 'my_testfile.nc.nc' -- you could test to see if the user has already supplied a .something at the end of the file: file_var.split('.')[-1]

Good point! Will fix.

Reply:
Based on my googling, I *think* it's:
```
import logging
import sys

logging.basicConfig(stream=sys.stdout, level=logging.INFO)
logger = logging.getLogger('LOGGER_NAME')
logger.info('shown on the console')

```
I don't know for sure if that's right, and I don't really get why it's such a big deal anyway when we're developing for the Jupyter notebook platform, and seems like something that could be changed later, so I wouldn't have it block merging.

Reply:
oh okay! My googling failed me. Will give this a try, thank you!

Reply:
oh I see - it's saving as a variable attribute rather than a global one. Huh.